distributed/scheduler.py

```diff
@@ -7243,6 +7244,27 @@ def set_restrictions(self, comm=None, worker=None):
                 restrictions = {restrictions}
             ts._worker_restrictions = set(restrictions)
 
+    def get_task_prefixes(self, comm=None):
+        with log_errors():
+            state = {
+                "memory": {},
+                "erred": {},
+                "released": {},
+                "processing": {},
+                "waiting": {},
+            }
+
+            for tp in self.task_prefixes.values():
+                active_states = tp.active_states
+                if any(active_states.get(s) for s in state.keys()):
+                    state["memory"][tp.name] = active_states["memory"]
+                    state["erred"][tp.name] = active_states["erred"]
+                    state["released"][tp.name] = active_states["released"]
+                    state["processing"][tp.name] = active_states["processing"]
+                    state["waiting"][tp.name] = active_states["waiting"]
+
+        return state
```

Comment:
Naively I would have expected a method named `get_task_prefixes` to return an iterable of prefix names. Maybe `get_task_prefix_states` would be more clear?

Reply:
(Not meant to be a blocking comment) Also, other similar methods (e.g. `get_task_status`) return a dictionary with task / task prefix names for keys. If it's not overly inconvenient for some other reason, it would be nice to do the same thing here. For example, instead of returning 

```python
{"memory": {"inc": 12, "double": 47, ...},
 "erred":  {"inc": 0, "double": 3, ...},
 ...
}
```

return 

```python
{"inc": {"memory": 12, "erred": 0, ...},
 "double":  {"memory": 47, "erred": 3, ...},
 ...
}
```